storm-core/src/clj/backtype/storm/ui/core.clj

```diff
@@ -843,14 +870,21 @@
     (populate-context! servlet-request)
     (assert-authorized-user "getClusterInfo")
     (json-response (nimbus-summary) (:callback m)))
+  (GET "/api/v1/history/summary" [:as {:keys [cookies servlet-request]} & m]
+    (let [user (.getUserName http-creds-handler servlet-request)]
+      (json-response (topology-history-info user) (:callback m))))
   (GET "/api/v1/supervisor/summary" [:as {:keys [cookies servlet-request]} & m]
-    (populate-context! servlet-request)
-    (assert-authorized-user "getClusterInfo")
-    (json-response (supervisor-summary) (:callback m)))
+    (assert-authorized-user servlet-request "getClusterInfo")
```

Comment:
This needs to stay as populate-context! before calling assert-authorized-user, or the code will fail when security is enabled.